### PR TITLE
Anton/elo voting

### PR DESCRIPTION
**Pull request by antonjb**
The frontend functionality for Elo Pair voting. This doesn't include any of the design, just fetching a pair and posting the votes to the backend.

### Test Plan

- [ ]  Comment out the date checking logic in `getServerSideProps`
- [ ] Visit `vote/elo`
- [ ] Cast votes, submit draw and generally use the feature
- [ ] You should see the appropriate network activity

### Device and Browser Testing

* Firefox

### CODE REVIEW

File: .env.local [hunk 1]
```diff
@@ -3,3 +3,5 @@ NEXT_PUBLIC_GET_SUBMISSIONS_URL=https://dddperth-functions-test.azurewebsites.ne
 NEXT_PUBLIC_SUBMIT_VOTE_URL=https://dddperth-functions-test.azurewebsites.net/api/SubmitVote
 NEXT_PUBLIC_GET_AGENDA_URL=https://dddperth-functions-test.azurewebsites.net/api/GetAgenda
 NEXT_PUBLIC_SUBMIT_FEEDBACK_URL=https://dddperth-functions-test.azurewebsites.net/api/SubmitFeedback
+NEXT_PUBLIC_ELO_PAIR=https://dddperth-functions-test.azurewebsites.net/api/EloVotingGetPair
+NEXT_PUBLIC_ELO_VOTE=https://dddperth-functions-test.azurewebsites.net/api/EloVotingSubmitPair
```
Comment (antonjb) [inline .env.local:7]: @robdmoore does anything need to happen to support new environment variables

File: pages/vote/elo.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+          Neither
+        </Button>
+      </div>
+    </Main>
+  )
+}
+
+export async function getServerSideProps() {
+  const dates = getConferenceDates(Conference, dateTimeProvider.now())
+  if (!dates.VotingOpen) {
```
Comment (antonjb) [inline pages/vote/elo.tsx:104]: Comment this out for testing

**Review by mzaatar — APPROVED** [review]
Looks good. Let's see it in test

**Comment by robdmoore** [comment]
Yes.

They need to be added to the variable libraries in Azure DevOps

--



On 7 Apr 2022, at 10:00 pm, Anton ***@***.***> wrote:

﻿

@antonjb commented on this pull request.

________________________________

In .env.local<https://github.com/dddwa/dddperth-website/pull/393#discussion_r845049195>:

> @@ -3,3 +3,5 @@ NEXT_PUBLIC_GET_SUBMISSIONS_URL=https://dddperth-functions-test.azurewebsites.ne
 NEXT_PUBLIC_SUBMIT_VOTE_URL=https://dddperth-functions-test.azurewebsites.net/api/SubmitVote
 NEXT_PUBLIC_GET_AGENDA_URL=https://dddperth-functions-test.azurewebsites.net/api/GetAgenda
 NEXT_PUBLIC_SUBMIT_FEEDBACK_URL=https://dddperth-functions-test.azurewebsites.net/api/SubmitFeedback
+NEXT_PUBLIC_ELO_PAIR=https://dddperth-functions-test.azurewebsites.net/api/EloVotingGetPair
+NEXT_PUBLIC_ELO_VOTE=https://dddperth-functions-test.azurewebsites.net/api/EloVotingSubmitPair


@robdmoore<https://github.com/robdmoore> does anything need to happen to support new environment variables

**Pull request merged**